09_json/kasunfdo/cmd/puppy-server/main.go

```diff
@@ -0,0 +1,50 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+
+	"gopkg.in/alecthomas/kingpin.v2"
+
+	"github.com/anz-bank/go-course/09_json/kasunfdo/pkg/puppy"
+	"github.com/anz-bank/go-course/09_json/kasunfdo/pkg/puppy/store"
+)
+
+var out io.Writer = os.Stdout
+
+func main() {
+	var dataFile *os.File
+	kingpin.Flag("data", "path to puppy data file").Short('d').Default("data.json").FileVar(&dataFile)
+	_, err := kingpin.CommandLine.Parse(os.Args[1:])
+	if err != nil {
+		panic(err)
+	}
+
+	if err := createPuppies(dataFile, store.NewMapStore()); err != nil {
+		panic(err)
+	}
+}
+
+func createPuppies(dataFile io.Reader, store puppy.Storer) error {
+	data, err := ioutil.ReadAll(dataFile)
+	if err != nil {
+		return puppy.NewError(puppy.ErrInternal)
+	}
+	var puppies []puppy.Puppy
+	if err = json.Unmarshal(data, &puppies); err != nil {
+		return puppy.NewError(puppy.ErrInternal)
```

Comment:
Not sure if this should be an internal error. If JSON can't be marshalled it's more like "bad data format" or "bad user input", don't you think?

Reply:
Agree.